crates/bevy_picking/src/pointer.rs

```diff
@@ -29,41 +27,69 @@ use crate::backend::HitData;
 #[derive(Debug, Default, Clone, Copy, Eq, PartialEq, Hash, Component, Reflect)]
 #[require(PointerLocation, PointerPress, PointerInteraction)]
 #[reflect(Component, Default, Debug, Hash, PartialEq)]
-pub enum PointerId {
-    /// The mouse pointer.
-    #[default]
-    Mouse,
-    /// A touch input, usually numbered by window touch events from `winit`.
-    Touch(u64),
-    /// A custom, uniquely identified pointer. Useful for mocking inputs or implementing a software
-    /// controlled cursor.
-    #[reflect(ignore)]
-    Custom(Uuid),
+pub struct PointerId {
+    /// A stable identifier for a specific pointer.
+    ///
+    /// This is typically provided by winit,
+    /// and is unique for each touch input or mouse.
+    pub id: u64,
+    /// The type of hardware device that created this pointer.
+    pub kind: PointerKind,
 }
 
 impl PointerId {
+    /// The [`PointerId`] of the mouse.
+    ///
+    /// Currently, only one mouse pointer is supported.
+    pub const MOUSE: Self = Self {
+        id: 0,
+        kind: PointerKind::Mouse,
+    };
+
     /// Returns true if the pointer is a touch input.
     pub fn is_touch(&self) -> bool {
-        matches!(self, PointerId::Touch(_))
+        matches!(self.kind, PointerKind::Touch)
     }
     /// Returns true if the pointer is the mouse.
     pub fn is_mouse(&self) -> bool {
-        matches!(self, PointerId::Mouse)
+        matches!(self.kind, PointerKind::Mouse)
     }
-    /// Returns true if the pointer is a custom input.
-    pub fn is_custom(&self) -> bool {
-        matches!(self, PointerId::Custom(_))
+    /// Returns true if the pointer is a virtual input.
+    pub fn is_virtual(&self) -> bool {
+        matches!(self.kind, PointerKind::Virtual)
     }
     /// Returns the touch id if the pointer is a touch input.
     pub fn get_touch_id(&self) -> Option<u64> {
-        if let PointerId::Touch(id) = self {
-            Some(*id)
+        if let PointerKind::Touch = self.kind {
+            Some(self.id)
         } else {
             None
         }
     }
 }
 
+/// The input device that created a pointer.
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash, Reflect)]
+#[reflect(Default, Debug, Hash, PartialEq)]
+pub enum PointerKind {
+    /// A pointer that is controlled by a mouse.
+    ///
+    /// Not a rodent; the device that you move around on a desk.
+    Mouse,
```

Comment:
I think that `PointerKind` is meant to describe the behaviour of the pointer, not the actual input device, but it seems a bit unclear. It would be nice to be able to support things like a "mouse" pointer that's controlled by a joystick or a mouse controlled first person pov with diegetic touch screen UI.

Reply:
Yeah, the intent is that things like UI libraries will match on this variant, and behave accordingly.